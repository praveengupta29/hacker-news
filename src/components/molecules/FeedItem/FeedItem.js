import React from 'react';

import './FeedItem.scss';
import Anchor from '../../atoms/Anchor';
import { extractDomainFromUrl } from '../../../commons/utils/url';
import timeSince from '../../../utils/timeSince';
import HideFeed from '../../../containers/atoms/HideFeed/HideFeed';

const getWebsiteName = url => {
  const domain = extractDomainFromUrl(url);

  return domain ? (
    <div className="website-name secondary-color">({domain})</div>
  ) : (
    ''
  );
};

const FeedItem = ({ ...others }) => {
  const {
    author,
    created_at_i: timestampInSeconds,
    num_comments: numberOfComments,
    objectID,
    points,
    title,
    url,
  } = others;
  return (
    <div className="feed-item flex align-center">
      <div className="numeric-detail flex align-center">
        <div className="comments">{numberOfComments}</div>

        <div className="points flex">
          {points}

          <Anchor handleLinkClick={() => {}} className="up-vote-link">
            <div className="up-vote-arrow" />
          </Anchor>
        </div>
      </div>

      <div className="text-details lg-flex align-center">
        <div className="main-details lg-flex align-center">
          <div className="title">{title}</div>
          {getWebsiteName(url)}
        </div>

        <div className="other-details flex align-center">
          <span className="by-keyword secondary-color">by</span>

          <span className="author-name">{author}</span>

          <span className="time-since secondary-color">
            {timeSince(timestampInSeconds)}
          </span>

          <span className="bracket secondary-color">[</span>

          <HideFeed objectId={objectID} />

          <span className="bracket secondary-color">]</span>
        </div>
      </div>
    </div>
  );
};

export default FeedItem;
