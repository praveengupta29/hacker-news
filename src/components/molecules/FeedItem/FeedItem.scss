@import '../../../styles/utils/breakpoints';
@import '../../../styles/utils/colors';

.feed-item {
  font-size: 11px;
  padding: 6px 24px;
  @include tablet {
    padding-left: 32px;
  }
  @include desktop {
    padding-left: 48px;
  }

  .main-details,
  .title {
    margin-bottom: 5px;
    @include desktop {
      margin-bottom: 0;
    }
  }

  &:nth-child(2n) {
    background: $grey;
  }

  .comments {
    font-size: 13px;
    padding-right: 24px;
    width: 24px;
    @include tablet {
      padding-right: 32px;
    }
    @include desktop {
      padding-right: 40px;
    }
  }

  .points {
    font-size: 12px;
    padding-right: 8px;
    width: 45px;
  }

  .title {
    font-size: 14px;
    margin-right: 4px;
  }

  .website-name {
    margin: 0 4px 0 0;
  }

  .by-keyword {
    margin-right: 4px;
    opacity: 0.6;
  }

  .author-name {
    font-size: 12px;
    margin-right: 4px;
  }

  .time-since {
    margin-right: 4px;
  }
}
