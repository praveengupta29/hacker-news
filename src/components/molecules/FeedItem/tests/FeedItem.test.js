import React from 'react';
import { shallow } from 'enzyme';
import FeedItem from '../FeedItem';

describe('FeedItem component', () => {
  const props = {
    author: 'davidmakina',
    created_at_i: 1586685261,
    num_comments: 626,
    objectID: '787666',
    points: 656,
    title: 'The cost of virtualizing CS conferences',
    url: 'https://queue.acm.org/detail.cfm?id=1039523',
    voted: false,
  };
  const sinceAgoDate = '2020-04-13';

  it('should render correctly with `UpVote` and match snapshot', () => {
    const component = shallow(
      <FeedItem {...props} sinceAgoDate={sinceAgoDate} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with `UnVote` and match snapshot', () => {
    props.voted = true;

    const component = shallow(
      <FeedItem {...props} sinceAgoDate={sinceAgoDate} />,
    );
    expect(component).toMatchSnapshot();
  });
});
