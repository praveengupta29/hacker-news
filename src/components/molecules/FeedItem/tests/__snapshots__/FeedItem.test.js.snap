// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem component should render correctly with \`UnVote\` and match snapshot 1`] = `
<div
  className="feed-item flex align-center"
>
  <div
    className="numeric-detail flex align-center"
  >
    <div
      className="comments"
    >
      626
    </div>
    <div
      className="points flex"
    >
      656
    </div>
  </div>
  <div
    className="text-details lg-flex align-center"
  >
    <div
      className="main-details lg-flex align-center"
    >
      <div
        className="title"
      >
        The cost of virtualizing CS conferences
      </div>
      <div
        className="website-name secondary-color"
      >
        (
        queue.acm.org
        )
      </div>
    </div>
    <div
      className="other-details flex align-center"
    >
      <span
        className="by-keyword secondary-color"
      >
        by
      </span>
      <span
        className="author-name"
      >
        davidmakina
      </span>
      <span
        className="time-since secondary-color"
      >
        about 14 hours ago
      </span>
      <Connect(Vote)
        objectId="787666"
        voted={true}
      />
      <span
        className="bracket secondary-color"
      >
        [
      </span>
      <Connect(HideFeed)
        objectId="787666"
      />
      <span
        className="bracket secondary-color"
      >
        ]
      </span>
    </div>
  </div>
</div>
`;

exports[`FeedItem component should render correctly with \`UpVote\` and match snapshot 1`] = `
<div
  className="feed-item flex align-center"
>
  <div
    className="numeric-detail flex align-center"
  >
    <div
      className="comments"
    >
      626
    </div>
    <div
      className="points flex"
    >
      656
      <Connect(Vote)
        objectId="787666"
      />
    </div>
  </div>
  <div
    className="text-details lg-flex align-center"
  >
    <div
      className="main-details lg-flex align-center"
    >
      <div
        className="title"
      >
        The cost of virtualizing CS conferences
      </div>
      <div
        className="website-name secondary-color"
      >
        (
        queue.acm.org
        )
      </div>
    </div>
    <div
      className="other-details flex align-center"
    >
      <span
        className="by-keyword secondary-color"
      >
        by
      </span>
      <span
        className="author-name"
      >
        davidmakina
      </span>
      <span
        className="time-since secondary-color"
      >
        about 14 hours ago
      </span>
      <span
        className="bracket secondary-color"
      >
        [
      </span>
      <Connect(HideFeed)
        objectId="787666"
      />
      <span
        className="bracket secondary-color"
      >
        ]
      </span>
    </div>
  </div>
</div>
`;
